wit/consumer.wit

```diff
@@ -1,17 +1,19 @@
+/// The consumer interface allows a guest to dynamically update its subscriptions and configuration
+/// as well as functionality for completing (acking) or abandoning (nacking) messages.
 interface consumer {
-    // {client, message, channel, error, guest-configuration}
-    use messaging-types.{client, message, channel, error, guest-configuration};
-
-    /// Blocking receive for t-milliseconds with ephemeral subscription – if no message is received, returns None
-    subscribe-try-receive: func(c: client, ch: channel, t-milliseconds: u32) -> result<option<list<message>>, error>;
-
-    /// Blocking receive until message with ephemeral subscription
-    subscribe-receive: func(c: client, ch: channel) -> result<list<message>, error>;
+    use types.{client, message, channel, error, guest-configuration};
 
     /// 'Fit-all' type function for updating a guest's configuration – this could be useful for:
     ///     - unsubscribing from a channel,
     ///     - checkpointing,
     ///     - etc..
+    /// 
+    /// Please note that implementations that provide `wasi:messaging` are responsible for ensuring
+    /// that guests are not allowed to subscribe to channels that they are not configured to
+    /// subscribe to (or have access to). Failure to do so can result in possible breakout or access
+    /// to resources that are not intended to be accessible to the guest. This means implementations
+    /// should validate that the configured topics are valid topics the guest should have access to or 
+    /// enforce it via the credentials used to connect to the service.
     update-guest-configuration: func(gc: guest-configuration) -> result<_, error>;
```

Comment:
What is the rationale behind leaving this function in the interface considering your comment in `guest.wit`? 

> "Configuration should be provided by the host to the guest via runtime-config or be handled by the messaging implementation itself. This matches how other things are done right now as well in keyvalue"

For clarification - Prior, I had imagined people using this interface to do one of the following:
(1) call `configure` from `guest.wit` to setup a `handler` to receive messages for that specific configuration, or
(2) use the constructs in `consumer.wit` to receive messages from non-commital/ephemeral subscriptions, which was to fulfill the `request-reply` pattern.

W/ adding runtime-config into the picture - how are people expected to set the channel(s) that the handler should be listening to? Plus - isn't runtime-config rn only capable of `get`ting configs? Or, is the idea that the handler will receive all messages and it is the implementor's job to do a `match` on channels (this could be annoying for users because `channel` is just a wrapper for `string`)? If the latter is correct, the design is changing signficantly, because, I had sort of imagined people creating multiple handler implementations for different configs.

Reply:
So I left this in after a conversation with @devigned about it. The whole purpose was to handle dynamically indicating to a host that you wanted to update the channels being passed to your handle function especially in cases where you are reacting to another message you've received. I think he can give a better rundown here. 

As for the matching, that is what I intended. I definitely was thinking that a handler function would have a `match` on channels (optionally) if you are subscribing to multiple, or you could just handle each message no matter what the channel. That seems to be a common way to do things in my experience. However, as for setting the channels, that is something that seems solidly in the realm of the host/person running and operating a component. They would say when they start it "please pass all messages from this channel or list of channels to this component." If the component needs to know those channel names, the host can provide them via runtime config. This is also important because otherwise some of the config details could very much be specific to implementations, which should be set outside of the component. Otherwise you end up with the same coupling that exists now

I am curious though how this would work:

> I had sort of imagined people creating multiple handler implementations for different configs

You can only export one `handle` function (or one per aliased interface, e.g. `export handler as another_handler`). So I wasn't sure if you meant that someone would create different components with baked in configs?

Reply:
> As for the matching, that is what I intended. I definitely was thinking that a handler function would have a match on channels (optionally) if you are subscribing to multiple, or you could just handle each message no matter what the channel. That seems to be a common way to do things in my experience.

Ok - If that is how we want to move forward, I would probably just add to the description of the function that we have the expectation that an implementor will `match` or figure out channels themselves.

> However, as for setting the channels, that is something that seems solidly in the realm of the host/person running and operating a component.

About channels, I have always seen it as a consumer responsibility to handle subscriptions - but this may be biased on my use of Service Bus, and MQTT. Also - there is the concept of ephemeral subscriptions to channels too w/ things like `subscribe-try-receive` where a consumer gets a message from a channel and the subscription is scoped only for that one function call. Although, I suppose that doesn't really fit within the `handler` model all that much.

> You can only export one handle function (or one per aliased interface, e.g. export handler as another_handler). So I wasn't sure if you meant that someone would create different components with baked in configs?

This bit was inspired in the idea of maintaining a component's lifetime as short as possible. That is, if we have one handler that manages all channels, the process will probably be very long-lived, which I was trying to avoid. Back in SpiderLightning, the way we added multiple handlers per HTTP endpoint was with a [custom macro](https://github.com/deislabs/spiderlightning/blob/283caa0299750028a0955d1702e8c24d62c415a1/crates/http-handler-macro/src/lib.rs#L26-L54), but that is obviously not portable as it is. So, like you mentioned, this could be done with aliased interfaces - w/ how the interface was previously, you'd have to alias two functions: `configure`, and `handler`. This way, `configure_sasquatch ` would set the subscriptions for `handler_sasquatch`, and so forth.